src/android/CodePushReportingManager.java

```diff
@@ -48,7 +48,7 @@ public void reportStatus(Status status, String label, String appVersion, String
         /* JS function to call: window.codePush.reportStatus(status: number, label: String, appVersion: String, currentDeploymentKey: String, previousLabelOrAppVersion?: string, previousDeploymentKey?: string) */
         final String script = String.format(
             Locale.US,
-            "javascript:window.codePush.reportStatus(%d, %s, %s, %s, %s, %s)",
+            "javascript:document.addEventListener(\"deviceready\", reportStatus); function reportStatus() { window.codePush.reportStatus(%d, %s, %s, %s, %s, %s); }",
```

Comment:
Would it be cleaner to do `addEventListener("deviceready", codePush.reportStatus.bind(codePush, %d, %s, %s, %s, %s)`? That way you don't need the extra inline method definition?